src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -52,28 +52,71 @@ export interface AuthenticationResult {
     message?: string;
 }
 
+export interface AuthenticationContext {
+    state?: unknown;
+    interactiveRequest: InteractiveAuthenticationRequest;
+}
+
+export interface InteractiveAuthenticationRequest {
+    scopes?: string[];
+    additionalRequestParameters?: { [key: string]: any };
+};
+
 export interface AuthorizeService {
     getUser(): Promise<unknown>;
     getAccessToken(request?: AccessTokenRequestOptions): Promise<AccessTokenResult>;
-    signIn(state: unknown): Promise<AuthenticationResult>;
+    signIn(context: AuthenticationContext): Promise<AuthenticationResult>;
     completeSignIn(state: unknown): Promise<AuthenticationResult>;
-    signOut(state: unknown): Promise<AuthenticationResult>;
+    signOut(context: AuthenticationContext): Promise<AuthenticationResult>;
     completeSignOut(url: string): Promise<AuthenticationResult>;
 }
 
+export class ManagedLogger {
+    private debug: boolean;
+    private trace: boolean;
+    public constructor(private objectRef: any) {
+        this.debug = objectRef.invokeMethod('IsEnabled', LogLevel.Debug);
+        this.trace = objectRef.invokeMethod('IsEnabled', LogLevel.Trace);
+    }
+
+    log(level: LogLevel, message: string): void {
+        if ((level == LogLevel.Trace && this.trace) ||
+            (level == LogLevel.Debug && this.debug)) {
+            return this.objectRef.invokeMethod('Log', level, message);
+        }
+    }
+}
+
+// These are the values for the .NET logger LogLevel
+export enum LogLevel {
+    Trace = 0,
+    Debug = 1,
+    Information = 2,
+    Warning = 3,
+    Error = 4,
+    Critical = 5,
+}
+
```

Comment:
I have been adding logging as even though is "orthogonal", it makes it much easier to debug/troubleshoot things.